src/pytest_inmanta_lsm/plugin.py

```diff
@@ -122,26 +148,42 @@ def remote_orchestrator_settings() -> Dict[str, Union[str, int, bool]]:
 
 
 @pytest.fixture
-def remote_orchestrator(project: Project, request, remote_orchestrator_settings) -> Iterator[RemoteOrchestrator]:
+def remote_orchestrator(
+    project: Project,
+    request: pytest.FixtureRequest,
+    remote_orchestrator_settings: Dict[str, Union[str, int, bool]],
+    docker_orchestrator: Optional[DockerOrchestrator],
+    remote_orchestrator_environment: str,
+    remote_orchestrator_noclean: bool,
+    remote_orchestrator_host: Tuple[str, int],
+) -> Iterator[RemoteOrchestrator]:
     LOGGER.info("Setting up remote orchestrator")
 
-    env = get_opt_or_env_or(request.config, "inm_lsm_env", "719c7ad5-6657-444b-b536-a27174cb7498")
-    host = get_opt_or_env_or(request.config, "inm_lsm_remote_host", "127.0.0.1")
-    user = get_opt_or_env_or(request.config, "inm_lsm_remote_user", "centos")
-    port = get_opt_or_env_or(request.config, "inm_lsm_remote_port", "22")
-    noclean = get_opt_or_env_or(request.config, "inm_lsm_noclean", "false").lower() == "true"
-    container_env = get_opt_or_env_or(request.config, "inm_lsm_container_env", "false").lower() == "true"
-    ssl = get_opt_or_env_or(request.config, "inm_lsm_ssl", "false").lower() == "true"
-    token = get_opt_or_env_or(request.config, "inm_lsm_token", None)
-    ca_cert = get_opt_or_env_or(request.config, "inm_lsm_ca_cert", None)
+    host, port = remote_orchestrator_host
 
-    if ssl:
-        if not os.path.isfile(ca_cert):
-            raise FileNotFoundError(f"Invalid path to CA certificate file: {ca_cert}")
-        ca_cert = os.path.abspath(ca_cert)
+    if docker_orchestrator is None:
+        ssh_user = inm_lsm_ssh_user.resolve(request.config)
+        ssh_port = str(inm_lsm_ssh_port.resolve(request.config))
+        container_env = inm_lsm_container_env.resolve(request.config)
     else:
-        if ca_cert:
-            LOGGER.warning("ssl option is set to False, so the CA certificate won't be used")
+        # If the orchestrator is running in a container we deployed ourself, we overwrite
+        # a few configuration parameters with what matches the deployed orchestrator
+        # If the container image behaves differently than assume, those value won't work,
+        # no mechanism exists currently to work around this.
+        ssh_user = "inmanta"
+        ssh_port = "22"
+        container_env = True
+
+    ssl = inm_lsm_ssl.resolve(request.config)
```

Comment:
SSL or AUTH + Docker will never work I think?

Perhaps raise a warning?

Reply:
I added a warning if ssl is set, the orchestrator container is used and the docker-compose file is not overwritten